### DISCUSSION TITLE
Is there a way to programmatically answer prompts?

### DISCUSSION THREAD

**Discussion by cswilliams** [piotrmurach/tty-prompt#166] [Q&A] [answered]
I'd like to add a `--yes` option to my program where it will automatically answer "yes" to any yes/no prompts.

Is there any supported way in the library to do this?  I realize I can just skip over the prompt entirely, but it still would be nice if the output on the screen looked the same as if the user answered "yes".  I came up with something like this, but was wondering if there was a better way?  Thanks!!

```ruby
require 'tty-prompt'
prompt = TTY::Prompt.new(interrupt: :exit)
question = "Do you like Ruby?"
yes = true if ARGV[0] == "--yes"
color = Pastel.new

if yes
  puts "#{question} #{color.green("Yes")}"
else
  prompt.yes?(question)
end
```

**Comment by piotrmurach** [comment, marked as answer]
Hi Chris,

I like your solution, it's neat. Another approach could be to provide a 'scripted' input to the prompt like so:

```ruby
if yes
  input = StringIO.new
  input << "y\n"
  input.rewind
  prompt = TTY::Prompt.new(input: input)
end

prompt.yes?("Do you like Ruby?")
```

**Reply by pboling** [reply]
I used this approach to test the `service_actor-promptable` gem.  It works well!
